topics/go/algorithms/data/bloom/bloom.go

```diff
@@ -0,0 +1,78 @@
+package bloom
+
+import (
+	"hash/maphash"
+	"math/big"
+	"math/rand"
+)
+
+// Bloom is an implementation of a bloom filter.
+type Bloom struct {
+	m    int      // Number of bits
+	k    int      // Number of probes
+	mask *big.Int // Underlying bitmask
+
+	hash maphash.Hash // Used to generate seed for key
+	r    *rand.Rand   // Used to generate "k" bit numbers
+}
+
+// New creates a new Bloom filter with "m" bits and "k" has functions.
+func New(m, k int) *Bloom {
+	b := Bloom{
+		m:    m,
+		k:    k,
+		r:    rand.New(rand.NewSource(0)),
+		mask: big.NewInt(0),
+	}
+	return &b
+}
+
```

Comment:
Let's pull all the exported functions before the unexported functions. It helps in seeing what the public API is as opposed to what is private.